plenum/server/primary_selector.py

```diff
@@ -1,41 +1,279 @@
+from typing import Iterable, List
+
+from plenum.common.types import ViewChangeDone
+from plenum.server.router import Route
 from stp_core.common.log import getlogger
-from plenum.server import replica
 from plenum.server.primary_decider import PrimaryDecider
+from plenum.server.replica import Replica
+from plenum.common.util import mostCommonElement, get_strong_quorum
 
 logger = getlogger()
 
 
-# TODO: Assumes that all nodes are up. Should select only
-# those nodes which are up
 class PrimarySelector(PrimaryDecider):
+    """
+    Simple implementation of primary decider. 
+    Decides on a primary in round-robin fashion.
+    Assumes that all nodes are up
+    """
+
     def __init__(self, node):
         super().__init__(node)
-        self.nodeNamesByRank = sorted(self.nodeNames)
+        self.previous_master_primary = None
+        self._view_change_done = {}  # instance id -> replica name -> data
+        self._ledger_manager = self.node.ledgerManager
 
-    def decidePrimaries(self):  # overridden method of PrimaryDecider
-        self.scheduleSelection()
+    @property
+    def routes(self) -> Iterable[Route]:
+        return [(ViewChangeDone, self._processViewChangeDoneMessage)]
```

Comment:
Yes, but round-robin-like primary decider can be just a very simple class that returns the next primary for an instance given a view change number as an input. We can use this simple class in multiple implementations of view change protocol (regardless ViewChangeDone-specific logic).
I think no need to do this re-factoring now, let's just put a TODO comment. We already have a ticket for this improvement: https://jira.hyperledger.org/browse/INDY-235